ethereum/circuits/lib/src/verifiers/storage.nr

```diff
@@ -1,17 +1,20 @@
-use dep::proof::trie_proof::TrieProof;
-
 use crate::misc::types::Bytes32;
-use crate::account_with_storage::StorageProof;
-
-pub fn verify_storage_value(storage_root: Bytes32, proof: StorageProof) {
-    let trie_proof = TrieProof { key: proof.key, value: proof.value, proof: proof.proof, depth: proof.depth };
-
-    // Method verify_storage_root asserts on failure or returns true; it doesn't return false. Validating return for safety.
-    assert(trie_proof.verify_storage_root(storage_root), "TrieProof: Invalid storage root");
-}
+use crate::account::MAX_PREFIXED_KEY_NIBBLE_LEN;
+use crate::account_with_storage::{MAX_STORAGE_VALUE_LEN, MAX_STORAGE_LEAF_LEN, LegacyStorageProof};
+use crate::merkle_patricia_proofs::proof::{ProofInput, verify_merkle_proof};
 
-pub fn verify_storage_values<N>(storage_root: Bytes32, proofs: [StorageProof; N]) {
+pub fn verify_storage_values<N, MAX_STORAGE_DEPTH_NO_LEAF>(
+    proof_inputs: [ProofInput<MAX_PREFIXED_KEY_NIBBLE_LEN, MAX_STORAGE_VALUE_LEN, MAX_STORAGE_DEPTH_NO_LEAF, MAX_STORAGE_LEAF_LEN>; N],
+    storage_root: Bytes32
+) {
     for i in 0..N {
-        verify_storage_value(storage_root, proofs[i]);
+        // verify_storage_value(storage_root, proofs[i]);
```

Comment:
?

Reply:
Sorry, this line should be removed as we can easily verify all storage values at once in the loop